cli/azd/cmd/init.go

```diff
@@ -374,10 +380,44 @@ func (i *initAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 
 func (i *initAction) initAppWithAgent(ctx context.Context) error {
 	// Warn user that this is an alpha feature
-	i.console.WarnForFeature(ctx, llm.FeatureLlm)
+	i.console.MessageUxItem(ctx, &ux.MessageTitle{
+		Title: fmt.Sprintf("Agentic mode init is in alpha mode. The agent will scan your repository and "+
+			"attempt to make an azd-ready template to init. You can always change permissions later "+
+			"by running `azd mcp consent`. Mistakes may occur in agent mode. "+
+			"To learn more, go to %s\n", output.WithLinkFormat("https://aka.ms/azd-feature-stages")),
+		TitleNote: "CTRL C to cancel interaction \n? to pull up help text",
+	})
+
+	// Check read only tool consent
+	readOnlyRule, err := i.consentManager.CheckConsent(ctx,
+		consent.ConsentRequest{
+			ToolID:     "*/*",
+			ServerName: "*",
+			Operation:  consent.OperationTypeTool,
+			Annotations: mcp.ToolAnnotation{
+				ReadOnlyHint: common.ToPtr(true),
+			},
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	if !readOnlyRule.Allowed {
+		if err := i.console.DoInteraction(func() error {
+			consentChecker := consent.NewConsentChecker(i.consentManager, "")
+			promptErr := consentChecker.PromptAndGrantReadOnlyToolConsent(ctx)
+			i.console.Message(ctx, "")
+
+			return promptErr
+		}); err != nil {
+			return err
+		}
+	}
 
 	azdAgent, err := i.agentFactory.Create(
 		agent.WithDebug(i.flags.global.EnableDebugLogging),
+		agent.WithFileWatching(true),
```

Comment:
Do we have a case where we don't want the file watching behavior? If not lets make it the default so that each caller doesn't need to set it.  Only need to set it when we want to turn it off.